testsuite/MDAnalysisTests/test_atomselections.py

```diff
@@ -788,3 +791,55 @@ def test_segid_and_resid():
     ref = ag.select_atoms('segid B').select_atoms('resid 1-100')
 
     assert_array_equal(ag.indices, ref.indices)
+
+
+class TestImplicitOr(object):
+    def setUp(self):
+        self.u = mda.Universe(PSF)
+
+    def tearDown(self):
+        del self.u
+
+    def _check_sels(self, ref, sel):
+        ref = self.u.select_atoms(ref)
+        sel = self.u.select_atoms(sel)
+
+        assert_array_equal(ref.indices, sel.indices)
+
+    def test_string_selections(self):
+        for ref, sel in (
+                ('name HT1 or name HT2 or name HT3', 'name HT1 HT2 HT3'),
+                ('type 2 or type 3 or type 4', 'type 2 3 4'),
+                ('resname MET or resname GLY', 'resname MET GLY'),
+                ('name H* or name N', 'name H* N'),
+                ('(name N or name HT1) and (resname MET or resname GLY)',
+                 'name N HT1 and resname MET GLY'),
+        ):
+            yield self._check_sels, ref, sel
```

Comment:
This is a [nosetest test generator](http://nose.readthedocs.org/en/latest/writing_tests.html#test-generators). It generates one test for each of the items.